torch/_dynamo/guards.py

```diff
@@ -719,6 +714,11 @@ def compile_check_fn(self, local_builder, global_builder):
                 tensor_check_names + ["tensor_check_names=tensor_check_names"]
             )
             verbose_code_parts.append(f"___check_tensors_verbose({verbose_args})")
+            if symbolic_shape_expression:
+                code_parts.append(symbolic_shape_expression)
+                verbose_code_parts.append(symbolic_shape_expression)
+                guards_out.add(Guard(name='symbolic_shape_expression', source=None, create_fn=self._parse_symbolic_shape_expressions, code_list=symbolic_shape_expression))
+
```

Comment:
Moving this here allows us to remove the tensor isinstance check, as the tensor check above already has a superset of it.